`System.IO.IOException: Sharing violation on path` executing `dotnet publish` when referencing package containing culture specific resource assemblies in subdirectories

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨 -->

<!--
Please fill in the *entire* template below.
The template is designed to avoid unnecessary delays to confirm and fix reported issues. Issues with missing information generally end up with a `need-info` label and take significantly more time to fix. Please help us help you!
For more information see https://github.com/xamarin/xamarin-macios/wiki/Submitting-Bugs-&-Suggestions
For support requests use https://www.xamarin.com/support
-->

### Steps to Reproduce

I have a `net6.0-ios` targeting project, that references a package that contains culture specific resource assemblies (the package is a locally build `Xamarin.Forms` package, of which the iOS related parts target `net6.0-ios`).

I am running the following build command for my project on my Windows 10 development machine, to build the project remotely on my macOS build host:

`dotnet publish "C:\PathToSolution\src\ProjectName\ProjectName.csproj" -c Release -p:ServerAddress=192.168.0.3 -p:ServerUser=username -bl -v d`

This results in the following build errors:

```
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/sv/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/ar/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/th/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/ca/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/tr/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/cs/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/uk/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/da/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/vi/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/de/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/zh-HK/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/el/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/es/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/zh-Hans/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/fi/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/zh-Hant/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/fr/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/he/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/hi/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/hr/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/hu/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/it/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/ja/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/ko/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/ms/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/nb/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/nl/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/pl/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/pt-BR/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/pt/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/ro/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/ru/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/sk/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
```

Looks to me, like the resource assemblies that are located in culture specific subdirectories on my Windows 10 development machine (e.g. `.../ca/Xamarin.Forms.Platform.iOS.resources.dll`) are all tried to be accessed on the macOS build host from the same directory (`...\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll`).

### Expected Behavior

The `Xamarin.MacDev.Tasks.ILStrip` task in the `_StripAssemblyIL` target should run successfully.

### Actual Behavior

A `System.IO.IOException: Sharing violation on path /Users/username/...\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll` exception is thrown for each culture specific resource assembly.

### Environment

<!--
1.
Visual Studio: Help > About Microsoft Visual Studio > Copy Info [button]
Visual Studio for Mac: Visual Studio > About Visual Studio > Show Details > Copy Information [button]
2. Paste into the code block below (between ```)
-->

```
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100\
```

### Build Logs

<!--
1. Place cursor below this comment block.
2. Attach build log or link to gist (https://gist.github.com/) of the log.
3. Logs can be found in the Build Output tab of the Errors pad in Visual Studio
To get full build logs, set the log verbosity to diagnostic at the following locations:
- On Visual Studio for Mac: Preferences > Projects > Build
- On Visual Studio for Windows: Tools > Options > Projects and Solutions > Build and Run
On Visual Studio for Windows you also want to add `-v -v -v -v` to the mtouch additional arguments by right-clicking the project in the solution explorer and selecting `Properties`.
-->

<details>
<summary>Relevant build log excerpt with <i>--verbosity detailed</i></summary>

```
         ILStrip: 2022-01-21T16:44:31.9130119+01:00 - Started
         ILStrip: 2022-01-21T16:44:31.9130805+01:00 - Initializing
         [xma]: Trying to get a Build Connection for Session 'c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d': Xamarin.Messaging.Build.Client.BuildConnection.c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d, Lifetime: Build
         ILStrip: 2022-01-21T16:44:31.9132159+01:00 - Initialized
         ILStrip: 2022-01-21T16:44:31.9138527+01:00 - There's no available inputs to copy to the Mac
         ILStrip: 2022-01-21T16:44:31.9138656+01:00 - Serializing intputs
         ILStrip: 2022-01-21T16:44:31.9303601+01:00 - Executing
         [xma]: Starting remote task execution for 'ProjectName': Xamarin.MacDev.Tasks.ILStrip
         [xma]: Sending Request Xamarin.Messaging.Build.Contracts.ExecuteTaskMessage to topic xvs/build/execute-task/ProjectName/c42c14e002fILStrip
         [xma]: Received Response of Xamarin.Messaging.Build.Contracts.ExecuteTaskMessage to topic buildc42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d22556Username/+/xvs/build/execute-task/ProjectName/c42c14e002fILStrip
         ILStrip: 2022-01-21T16:44:33.0852751+01:00 - Logging messages
         [ILStrip] C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/id/Xamarin.Forms.Platform.iOS.resources.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
         [ILStrip] C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/ar/Xamarin.Forms.Platform.iOS.resources.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
         [ILStrip] C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/sv/Xamarin.Forms.Platform.iOS.resources.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
         [ILStrip] obj/Release/net6.0-ios/ios-arm64/linked/System.Collections.NonGeneric.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\System.Collections.NonGeneric.dll
         [ILStrip] obj/Release/net6.0-ios/ios-arm64/linked/Microsoft.AppCenter.Crashes.iOS.Bindings.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\Microsoft.AppCenter.Crashes.iOS.Bindings.dll
         [ILStrip] obj/Release/net6.0-ios/ios-arm64/linked/System.Runtime.Serialization.Primitives.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\System.Runtime.Serialization.Primitives.dll
         [ILStrip] obj/Release/net6.0-ios/ios-arm64/linked/System.Drawing.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\System.Drawing.dll
         [ILStrip] obj/Release/net6.0-ios/ios-arm64/linked/System.Net.Security.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\System.Net.Security.dll
         System.IO.IOException: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
           at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00259] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/FileStream.cs:274
           at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/FileStream.cs:91
           at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
           at ILStrip+AssemblyStripper.StripAssembly (CilStrip.Mono.Cecil.AssemblyDefinition assembly, System.String file) [0x00000] in <f5c90e6d54eb4012b97c80398dbac604>:0
           at ILStrip.StripAssembly (Microsoft.Build.Framework.ITaskItem assemblyItem) [0x00065] in <f5c90e6d54eb4012b97c80398dbac604>:0
         System.IO.IOException: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
           at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00259] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/FileStream.cs:274
           at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/FileStream.cs:91
           at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
           at ILStrip+AssemblyStripper.StripAssembly (CilStrip.Mono.Cecil.AssemblyDefinition assembly, System.String file) [0x00000] in <f5c90e6d54eb4012b97c80398dbac604>:0
           at ILStrip.StripAssembly (Microsoft.Build.Framework.ITaskItem assemblyItem) [0x00065] in <f5c90e6d54eb4012b97c80398dbac604>:0
   1:7>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/id/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
   1:7>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/sv/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
         [ILStrip] C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/th/Xamarin.Forms.Platform.iOS.resources.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
         [ILStrip] C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/it/Xamarin.Forms.Platform.iOS.resources.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
         System.IO.IOException: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
           at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00259] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/FileStream.cs:274
           at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/FileStream.cs:91
           at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
           at ILStrip+AssemblyStripper.StripAssembly (CilStrip.Mono.Cecil.AssemblyDefinition assembly, System.String file) [0x00000] in <f5c90e6d54eb4012b97c80398dbac604>:0
           at ILStrip.StripAssembly (Microsoft.Build.Framework.ITaskItem assemblyItem) [0x00065] in <f5c90e6d54eb4012b97c80398dbac604>:0
   1:7>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.0.101-preview.11.551\targets\Xamarin.Shared.Sdk.targets(659,3): error : ILStrip failed for C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/th/Xamarin.Forms.Platform.iOS.resources.dll: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
         [ILStrip] C:/Users/Username/.nuget/packages/xamarin.forms/6.0.0/lib/Xamarin.iOS10/tr/Xamarin.Forms.Platform.iOS.resources.dll to obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
         System.IO.IOException: Sharing violation on path /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj\Release\net6.0-ios\ios-arm64\stripped\\Xamarin.Forms.Platform.iOS.resources.dll
           at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00259] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/FileStream.cs:274
           at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/FileStream.cs:91
           at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
           at ILStrip+AssemblyStripper.StripAssembly (CilStrip.Mono.Cecil.AssemblyDefinition assembly, System.String file) [0x00000] in <f5c90e6d54eb4012b97c80398dbac604>:0
           at ILStrip.StripAssembly (Microsoft.Build.Framework.ITaskItem assemblyItem) [0x00065] in <f5c90e6d54eb4012b97c80398dbac604>:0

    [...]
```

</details>

If necessary, I could privately share a binlog, but the excerpt should already contain all the relevant information regarding this issue.

I then temporarily removed the files to see whether the build succeed:

```xml
<Target Name="AddStripAssemblyILFix" AfterTargets="_ComputeStripAssemblyIL" BeforeTargets="_StripAssemblyIL">
    <ItemGroup>
        <CultureSpecificResourceAssembly Include="@(ResolvedFileToPublish)"
                                         Condition="'%(Filename)%(Extension)' == 'Xamarin.Forms.Platform.iOS.resources.dll'" />
        <ResolvedFileToPublish Remove="@(ResolvedFileToPublish)"
                               Condition="'%(Filename)%(Extension)' == 'Xamarin.Forms.Platform.iOS.resources.dll'" />
    </ItemGroup>
</Target>
<Target Name="RemoveStripAssemblyILFix" AfterTargets="_StripAssemblyIL">
    <ItemGroup>
        <ResolvedFileToPublish Include="@(CultureSpecificResourceAssembly)" />
    </ItemGroup>
</Target>
```

It doesn't. I now get the following error cluster:

```
C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.0.101-preview.11.551\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/Microsoft.AppCenter.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.0.101-preview.11.551\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/Microsoft.AppCenter.iOS.Bindings.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]

[...]
```

While the `obj/Release/net6.0-ios/ios-arm64/stripped` directory on my Windows 10 development machine contains the _empty placeholder files_ (as expected), the `/Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj/Release/net6.0-ios/ios-arm64/stripped` _directory itself_ on my macOS build host _is empty_ (seems unexpected).

### Issue/Erros with `15.2.100-preview.12.5` iOS workload

@rolfbjarne Using the `15.2.100-preview.12.5` iOS workload build against and as SDK for building (on both, the Windows 10 development machine and the macOS build host), I now get the following error cluster when executing `dotnet build -c Release`:

```
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/Newtonsoft.Json.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/Xamarin.Essentials.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/Xamarin.Forms.Platform.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/Xamarin.Forms.Platform.iOS.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/Xamarin.Forms.Xaml.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/System.ComponentModel.Primitives.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/System.ComponentModel.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/System.Drawing.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]

[...]

C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/System.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/mscorlib.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/netstandard.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/System.Private.CoreLib.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
C:\PathToSolution\.dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.100-preview.12.5\tools\msbuild\iOS\Xamarin.iOS.Common.After.targets(260,3): error MSB3030: Could not copy the file "obj/Release/net6.0-ios/ios-arm64//stripped/Xamarin.iOS.dll" because it was not found. [C:\PathToSolution\src\ProjectName\ProjectName.csproj]
```

While the empty file stubs exists on the Windows 10 development machine at `C:\PathToSolution\src\ProjectName\obj\Release\net6.0-ios\ios-arm64\stripped` (expected):

```
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----        26.01.2022     22:31                Microsoft.CSharp.dllProjects
d-----        26.01.2022     22:31                Microsoft.Win32.Primitives.dllProjects
d-----        26.01.2022     22:31                mscorlib.dllProjects
d-----        26.01.2022     22:31                netstandard.dllProjects
d-----        26.01.2022     22:31                Newtonsoft.Json.dllProjects
d-----        26.01.2022     22:31                System.Collections.Concurrent.dllProjects
d-----        26.01.2022     22:31                System.Collections.dllProjects
d-----        26.01.2022     22:31                System.Collections.NonGeneric.dllProjects
d-----        26.01.2022     22:31                System.Collections.Specialized.dllProjects

[...]
```

All of the **files** on the macOS build host are directly located at `/Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d` (unexpected):

```
~ % ls -lah /Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/
total 76488
drwxr-xr-x  153 username  staff   4.8K Jan 26 22:32 .
drwxr-xr-x    3 username  staff    96B Jan 26 22:32 ..
drwxr-xr-x   21 username  staff   672B Jan 26 22:32 Assets.xcassets
drwxr-xr-x    3 username  staff    96B Jan 26 22:32 C:
drwxr-xr-x    3 username  staff    96B Jan 26 22:32 E:
-rw-r--r--    1 username  staff   196B Jan 26 22:32 Entitlements.plist
drwxr-xr-x    3 username  staff    96B Jan 26 22:32 Resources
-rw-r--r--    1 username  staff    63B Jan 26 22:32 Roots.xml
drwxr-xr-x    4 username  staff   128B Jan 26 22:32 bin
drwxr-xr-x    4 username  staff   128B Jan 26 22:32 obj
-rw-r--r--    1 username  staff   123K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Microsoft.CSharp.dll
-rw-r--r--    1 username  staff   6.0K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Microsoft.Win32.Primitives.dll
-rw-r--r--    1 username  staff   396K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Newtonsoft.Json.dll
-rw-r--r--    1 username  staff    19K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Collections.Concurrent.dll
-rw-r--r--    1 username  staff    13K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Collections.NonGeneric.dll
-rw-r--r--    1 username  staff   8.0K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Collections.Specialized.dll
-rw-r--r--    1 username  staff    25K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Collections.dll
-rw-r--r--    1 username  staff   9.5K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.ComponentModel.Primitives.dll
-rw-r--r--    1 username  staff    68K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.ComponentModel.TypeConverter.dll
-rw-r--r--    1 username  staff   4.5K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.ComponentModel.dll
-rw-r--r--    1 username  staff   8.5K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Console.dll
-rw-r--r--    1 username  staff   4.0K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Core.dll
-rw-r--r--    1 username  staff   221K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Data.Common.dll
-rw-r--r--    1 username  staff   5.5K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Diagnostics.DiagnosticSource.dll
-rw-r--r--    1 username  staff    13K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Diagnostics.Process.dll
-rw-r--r--    1 username  staff    14K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Diagnostics.TraceSource.dll
-rw-r--r--    1 username  staff    27K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Drawing.Primitives.dll
-rw-r--r--    1 username  staff   4.5K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\System.Drawing.dll

[...]

-rw-r--r--    1 username  staff   123K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Xamarin.Essentials.dll
-rw-r--r--    1 username  staff   789K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Xamarin.Forms.Core.dll
-rw-r--r--    1 username  staff    12K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Xamarin.Forms.Platform.dll
-rw-r--r--    1 username  staff   370K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Xamarin.Forms.Platform.iOS.dll
-rw-r--r--    1 username  staff    67K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Xamarin.Forms.Xaml.dll
-rw-r--r--    1 username  staff    12M Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\Xamarin.iOS.dll
-rw-r--r--    1 username  staff   7.5K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\mscorlib.dll
-rw-r--r--    1 username  staff    18K Jan 26 22:32 obj\Release\net6.0-ios\ios-arm64\\stripped\netstandard.dll
```

As can be seen, the **files** have a _name_ that is exactly what the _full relative path_ of the items should have been.
So it appears that the files are wrongly named and reside at the wrong location.

### Workaround/Fix

I then applied the following **workaround/fix**, effectively replacing the original `_StripAssemblyIL` target:

```xml
<Target Name="After_ComputeStripAssemblyIL__Before_FixedStripAssemblyIL"
        Condition="'$(EnableAssemblyILStripping)' == 'true'"
        AfterTargets="_ComputeStripAssemblyIL"
        BeforeTargets="_StripAssemblyIL">
    <!-- Ensure that the real/broken _StripAssemblyIL target is not going to run, because we replace it
         with this target. -->
    <PropertyGroup>
        <EnableAssemblyILStripping>false</EnableAssemblyILStripping>
    </PropertyGroup>
    <!-- ^^^ -->

    <PropertyGroup>
        <_StrippedAssemblyDirectory>$(DeviceSpecificIntermediateOutputPath)\stripped</_StrippedAssemblyDirectory>
    </PropertyGroup>
    <ItemGroup>
        <!-- WORKAROUND: Ignore culture specific resource assemblies in subdirectories. -->
        <CultureSpecificResourceAssembly Include="@(ResolvedFileToPublish)" Condition="'%(Filename)%(Extension)' == 'Xamarin.Forms.Platform.iOS.resources.dll'" />
        <ResolvedFileToPublish Remove="@(ResolvedFileToPublish)" Condition="'%(Filename)%(Extension)' == 'Xamarin.Forms.Platform.iOS.resources.dll'" />
        <!-- ^^^ -->

        <_AssembliesToBeStripped Include="@(ResolvedFileToPublish)" Condition="'%(Extension)' == '.dll'">
            <OutputPath Condition="'%(ResolvedFileToPublish.DestinationSubPath)' != ''">$(_StrippedAssemblyDirectory)\%(ResolvedFileToPublish.DestinationSubPath)</OutputPath>
            <OutputPath Condition="'%(ResolvedFileToPublish.DestinationSubPath)' == ''">$(_StrippedAssemblyDirectory)\%(Filename)%(Extension)</OutputPath>
        </_AssembliesToBeStripped>

        <!-- FIX: Use forward slashes in item paths, otherwise ILStrip will create filenames that resemble the part of
                  the relative PATH of the item that uses backslashes, instead of writing the file to the LOCATION of the
                  relative path. -->
        <_AssembliesToBeStrippedTemp Include="@(_AssembliesToBeStripped)" />
        <_AssembliesToBeStripped Remove="@(ResolvedFileToPublish)" />
        <_AssembliesToBeStripped Include="@(_AssembliesToBeStrippedTemp->'%(Identity)'->Replace('\', '/'))">
            <OutputPath>$([System.String]::Copy('%(OutputPath)').Replace('\', '/'))</OutputPath>
        </_AssembliesToBeStripped>
        <!-- ^^^ -->

        <_AssemblyDirsToCreate Include="@(_AssembliesToBeStripped->'%(OutputPath)%(Directory)'->Distinct())"/>
    </ItemGroup>
    <MakeDir SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Directories="@(_AssemblyDirsToCreate)" />
    <Xamarin.MacDev.Tasks.ILStrip Assemblies="@(_AssembliesToBeStripped)" SessionId="$(BuildSessionId)">
        <Output TaskParameter="StrippedAssemblies" ItemName="_StrippedAssemblies" />
    </Xamarin.MacDev.Tasks.ILStrip>
    <ItemGroup>
        <ResolvedFileToPublish Remove="@(_AssembliesToBeStripped)" />
        <ResolvedFileToPublish Include="@(_StrippedAssemblies)" />
    </ItemGroup>
</Target>
```

Now the files are being stripped and placed correctly in `/Users/username/Library/Caches/Xamarin/mtbs/builds/ProjectName/c42c14ed284fb2fb3c932a4cc8d9e9afc0f6d0aaeb390b848220a469cd12041d/obj/Release/net6.0-ios/ios-arm64/stripped`.

The `\` occurences in paths had to be replaced with `/`, and of course my previous workaround to skip the culture specific resource assemblies in subdirectories had to be applied as well.

/cc @praeclarum

--- a/dotnet/targets/Xamarin.Shared.Sdk.targets
+++ b/dotnet/targets/Xamarin.Shared.Sdk.targets
@@ -699,6 +699,14 @@
 				<OutputPath Condition="'%(ResolvedFileToPublish.DestinationSubPath)' != ''">$(_StrippedAssemblyDirectory)\%(ResolvedFileToPublish.DestinationSubPath)</OutputPath>
 				<OutputPath Condition="'%(ResolvedFileToPublish.DestinationSubPath)' == ''">$(_StrippedAssemblyDirectory)\%(Filename)%(Extension)</OutputPath>
 			</_AssembliesToBeStripped>
+
+			<!-- Use forward slashes in OutputPath, otherwise ILStrip will create filenames that resemble the part of
+				 the relative path of the item that uses backslashes, instead of writing the file to the location of the
+				 relative path. -->
+			<_AssembliesToBeStripped>
+				<OutputPath>$([System.String]::Copy('%(OutputPath)').Replace('\', '/'))</OutputPath>
+			</_AssembliesToBeStripped>
+
 			<_AssemblyDirsToCreate Include="@(_AssembliesToBeStripped->'%(OutputPath)%(Directory)'->Distinct())"/>
 		</ItemGroup>
 		<MakeDir SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Directories="@(_AssemblyDirsToCreate)" />

